src/init.cpp

```diff
@@ -1208,8 +1209,11 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
         ShrinkDebugFile();
     }
 
-    if (fPrintToDebugLog)
-        OpenDebugLog();
+    if (fPrintToDebugLog) {
+        if (!OpenDebugLog()) {
+            return InitError(strprintf("Could not open debug log file %s", GetDebugLogPath().string()));
```

Comment:
Nit, missing test.

Reply:
Will add a test, though testing this particular message only makes sense after #11783 goes in

Reply:
Added test anyway, temporarily adding "Fix shutdown in case of errors during initialization" into here